pkg/3scale/amp/component/apicast.go

```diff
@@ -725,17 +695,18 @@ func (apicast *Apicast) productionDeploymentConfigAnnotations() map[string]strin
 		annotations[customEnvAnnotationKey(customEnvSecret)] = customEnvAnnotationValue(customEnvSecret)
 	}
 
-	// keep backward compat
-	if len(annotations) == 0 {
-		return nil
-	}
-
 	return annotations
 }
 
-func (apicast *Apicast) stagingDeploymentConfigAnnotations() map[string]string {
+func (apicast *Apicast) stagingDeploymentAnnotations() map[string]string {
 	annotations := map[string]string{}
 
+	// Image trigger annotation should always be present
+	imageTriggerString := reconcilers.CreateImageTriggerAnnotationString(map[string]string{
+		ApicastStagingName: fmt.Sprintf("amp-apicast:%v", apicast.Options.ImageTag),
+	})
+	annotations[reconcilers.DeploymentImageTriggerAnnotation] = imageTriggerString
```

Comment:
What is the purpose of this new annotation?

Reply:
This is needed for the Deployment objects to leverage the existing ImageStreams. See docs [here](https://docs.openshift.com/container-platform/4.14/openshift_images/triggering-updates-on-imagestream-changes.html) for further explanation.

Reply:
today I learned this from Openshift. Thanks

Have you considered not using triggers? What is the benefit you see in using triggers?

Reply:
My understanding is that we have to use triggers in order to use the existing ImageStreams. I'm fine with skipping triggers/ImageStreams but would we then need the operator to handle image versioning and patching the Deployments with each version upgrade?

Reply:
the operator is needed to handle image versioning and patching the deployments regardless of using imagestreams or not. 

The question is, do we want to keep using Imagestreams? and if so, what is the benefit?

Reply:
I don't have a strong opinion. ImageStreams contain some metadata, image history, etc. but not sure if that's good enough of a reason to keep them though.

Do you know what the original justification for using ImageStreams was? Was it just because DCs required them for triggers?

Reply:
@eguzki Would there ever be a scenario where a component's image would be updated outside of a 3scale-operator version bump? If so, then I think we need to continue to use ImageStreams to sync the updated component image to the Deployment, right?

Reply:
I do not think such scenario is realistic. It will not be supported use case. And for that, there are `Image` fields to update images using the APIMAnager CR.

I tend to think we can get rid of the imagestreams. It can be done in following up PRs

Reply:
Sounds good I'll merge this and will address it in another PR.